### PR TITLE
Move to 1.2.0-* versions 

### PR DESCRIPTION
**Pull request by prafullbhosale**
cc @mlorbetske

### CODE REVIEW

File: src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/PropertyMetadata.cs [hunk 1]
```diff
@@ -62,9 +63,11 @@ public PropertyMetadata(IProperty property, Type dbContextType)
 
             IsEnum = property.ClrType.GetTypeInfo().IsEnum;
             IsReadOnly = property.IsReadOnlyBeforeSave;
-
-            IsAutoGenerated = ((property.RequiresValueGenerator
-                || property.IsStoreGeneratedAlways)
+            
+            //TODO: RequiresValueGenerator() is an internal EF Core method.
+            // Need to confirm with EF if this is ok to use. 
+            IsAutoGenerated = ((property.RequiresValueGenerator()
```
Comment (prafullbhosale) [inline src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/PropertyMetadata.cs]: @bricelam @ajcvickers 
`property.RequiresValueGenerator()` is now in an internal namespace. 
Is it possible for EF to expose it as a public extension method?

File: NuGet.config [hunk 2]
```diff
@@ -5,6 +5,7 @@
     <add key="dotnet-cli" value="https://dotnet.myget.org/F/dotnet-cli/api/v3/index.json" />
     <add key="cli-deps" value="https://dotnet.myget.org/F/cli-deps/api/v3/index.json" />
     <add key="AspNetCore" value="https://dotnet.myget.org/F/aspnetcore-ci-dev/api/v3/index.json" />
+    <add key="AspNetCoreDev" value="https://dotnet.myget.org/F/aspnetcore-dev/api/v3/index.json" />
```
Comment (natemcmaster) [inline NuGet.config]: This shouldn't be necessary. These packages are just a later version of aspnetcore-ci-dev.

File: build/dependencies.props [hunk 3]
```diff
@@ -4,15 +4,15 @@
   </PropertyGroup>
   <!-- AspNetCore Versions -->
   <PropertyGroup>
-    <AspNetCoreHostingVersion>1.0.2</AspNetCoreHostingVersion>
-    <KestrelVersion>1.0.3</KestrelVersion>
-    <CommandLineUtilsVersion>1.0.1</CommandLineUtilsVersion>
+    <AspNetCoreHostingVersion>1.2.0-preview1-*</AspNetCoreHostingVersion>
```
Comment (natemcmaster) [inline build/dependencies.props]: Just `1.2.0-*` is fine for now.

File: test/Shared/MsBuildProjectStrings.cs [hunk 4]
```diff
@@ -14,6 +14,7 @@ public static string GetNugetConfigTxt(string artifactsDir)
     <packageSources>
         <clear />
         <add key=""local"" value=""" + artifactsDir +  @""" />
+        <add key=""AspNetCoreDev"" value=""https://dotnet.myget.org/F/aspnetcore-dev/api/v3/index.json"" />
```
Comment (natemcmaster) [inline test/Shared/MsBuildProjectStrings.cs]: Likewise, shouldn't be necessary.

File: build/dependencies.props [hunk 5]
```diff
@@ -4,15 +4,15 @@
...
-    <DependencyInjectionVersion>1.0.2</DependencyInjectionVersion>
-    <EntityFrameworkCoreVersion>1.0.3</EntityFrameworkCoreVersion>
-    <FileProvidersVersion>1.0.1</FileProvidersVersion>
-    <PlatformAbstractionsVersion>1.0.0</PlatformAbstractionsVersion>
-    <RazorVersion>1.0.2</RazorVersion>
+    <DependencyInjectionVersion>1.2.0-*</DependencyInjectionVersion>
+    <EntityFrameworkCoreVersion>1.2.0-*</EntityFrameworkCoreVersion>
+    <FileProvidersVersion>1.2.0-*</FileProvidersVersion>
+    <PlatformAbstractionsVersion>1.2.0-*</PlatformAbstractionsVersion>
+    <RazorVersion>1.2.0-*</RazorVersion>
```
Comment (pranavkm) [inline build/dependencies.props]: The Razor package you're referencing is going away. I'd recommend switching to Razor.Evolution.

Comment (prafullbhosale) [inline build/dependencies.props]: Filed a new issue for this: https://github.com/aspnet/Scaffolding/issues/421

**Comment by prafullbhosale** [comment]
@pranavkm if there are no more changes (other than #421) I'd like to merge this

File: build/dependencies.props (hunk 5 above)
Comment (pranavkm) [inline build/dependencies.props]: Most of the stuff from Razor is gone now - https://github.com/aspnet/Razor/commit/a7eb30ddca7872498696f2bb14c286105b0ae57c. So I'm not sure if this still compiles. cc @ryanbrandenburg

Comment (prafullbhosale) [inline build/dependencies.props]: You are right. It didn't compile, so I made the changes to move to `Razor.Evolution` instead.

File: src/Microsoft.VisualStudio.Web.CodeGeneration.Templating/RazorTemplating.cs [hunk 6]
```diff
@@ -5,7 +5,7 @@
 using System.IO;
 using System.Linq;
 using System.Threading.Tasks;
-using Microsoft.AspNetCore.Razor;
+using Microsoft.AspNetCore.Razor.Evolution;
```
Comment (prafullbhosale) [inline src/Microsoft.VisualStudio.Web.CodeGeneration.Templating/RazorTemplating.cs]: cc @pranavkm

**Pull request merged**